pkg/invariants/types.go

```diff
@@ -0,0 +1,43 @@
+package invariants
+
+import (
+	"context"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+)
+
+type InvariantTest interface {
+	// StartCollection is responsible for setting up all resources required for collection of data on the cluster.
+	// An error will not stop execution, but will cause a junit failure that will cause the job run to fail.
+	// This allows us to know when setups fail.
+	StartCollection(ctx context.Context, adminRESTConfig *rest.Config) error
+
+	// CollectData will only be called once.  Errors reported will be indicated as junit test failure and will
+	// cause job runs to fail.
+	CollectData(ctx context.Context) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error)
```

Comment:
At lot of invariant tests today require other intervals from the cluster to make complex decisions. For this reason it might be more futureproof to split this into CollectData() (Intervals, error) and RunTests(allClusterIntervals) ([]JunitTestCase, error).

Reply:
added